ionoscloud/config.go

```diff
@@ -16,22 +17,22 @@ type Config struct {
 	Token    string
 }
 
-// Client returns a new client for accessing profitbricks.
-func (c *Config) Client(terraformVersion string) (*profitbricks.Client, error) {
-	var client *profitbricks.Client
+// Client returns a new client for accessing ionoscloud.
+func (c *Config) Client(terraformVersion string) (*ionoscloud.APIClient, error) {
+	var client *ionoscloud.APIClient
 	if c.Token != "" {
-		client = profitbricks.NewClientbyToken(c.Token)
+		client = ionoscloud.NewAPIClient(ionoscloud.NewConfiguration("", "", c.Token))
 	} else {
-		client = profitbricks.NewClient(c.Username, c.Password)
+		client = ionoscloud.NewAPIClient(ionoscloud.NewConfiguration(c.Username, c.Password, ""))
 	}
-	client.SetUserAgent(httpclient.TerraformUserAgent(terraformVersion))
+	client.GetConfig().UserAgent = httpclient.TerraformUserAgent(terraformVersion)
 
-	log.Printf("[DEBUG] Terraform client UA set to %s", client.GetUserAgent())
+	log.Printf("[DEBUG] Terraform client UA set to %s", client.GetConfig().UserAgent)
 
-	client.SetDepth(5)
+	client.GetConfig().AddDefaultQueryParam("depth", strconv.Itoa(5))
```

Comment:
you don't need to `strconv.Itoa(5)` - you can just use the string "5" :)

Reply:
done